ocaml/typing/typeopt.ml

```diff
@@ -61,7 +61,7 @@ let scrape_poly env ty =
   | d -> d
 
 let is_function_type env ty =
-  match scrape env ty with
+  match scrape_poly env ty with
```

Comment:
Note: I'd like an input from someone knowledgeable of the typechecker to review this particular change, as it might impact other things.